### ISSUE TITLE
Dont starting Sway

### ISSUE THREAD

**Issue by Artlant** [SirCmpwn/sway#1174]
info:
```
~ >>> pacman -Qs xwayland
extra/xorg-server-xwayland 1.19.3-2 (xorg) [installed]
~ >>> uname -a                                                                                                                                            
Linux hostname 4.10.8-1-ARCH #1 SMP PREEMPT Fri Mar 31 16:50:19 CEST 2017 x86_64 GNU/Linux
~ >>> pacman -Qs sway          
sway 0.12.2-1
~ >>> pactree -d 1  sway           
sway
├─wlc
├─xorg-server-xwayland
├─json-c
├─pango
├─wayland
└─gdk-pixbuf2
```
     
logging include
[sway.log.txt](https://github.com/SirCmpwn/sway/files/923401/sway.log.txt)

config file is default 
Maybe you want to delete i3wm to work? But this did not require installation

**Comment by Artlant** [comment]
~ >>> ls -lt /usr/bin/sway                                                                                                                             10:07:58
-rwxr-xr-x 1 root root 291K Apr 12 20:42 /usr/bin/sway*

**Comment by Artlant** [comment]
~ >>> sudo sway                                                                                                                                       
04/15/17 10:12:08 - [main.c:45] [wlc] Failed to add socket to wayland display

**Comment by Artlant** [comment]
~ >>> cat /etc/pacman.d/hooks                                                                                                                          
[Trigger]
Operation = Install
Type = Package
.....
Exec = /usr/bin/chmod +s /usr/bin/sway

**Comment by Artlant** [comment]
Why? Why not start?

**Comment by Artlant** [comment]
new log
```
04/15/17 10:16:54 - [main.c:49] [wlc] error: XDG_RUNTIME_DIR not set in the environment
04/15/17 10:16:54 - [main.c:45] [wlc] Failed to add socket to wayland display
04/15/17 10:16:54 - [main.c:49] [wlc] Parent exit (25325)
```
[sway.log.txt](https://github.com/SirCmpwn/sway/files/923405/sway.log.txt)

**Comment by JerziKaminsky** [comment]
- You don't need to delete i3
- Use `sway -d` to get a more detailed log
- Tell us what wlc version you're using
- Tell us what distro you're using

When you use sudo it executes the program with a sanitized environment as a security measure. Looks like XDG_RUNTIME_DIR is lost, so sway/wlc can't locate the wayland socket that lives there in order to connect to the server.

Try running without sudo first, then sett sway to setuid (it carefully drops privileges shortly after starting).

... and try to post a single detailed report instead of 6 within 5 minutes, it makes a bad impression. :)

Hope this helps.

**Comment by Artlant** [comment]
"Use `sway -V` or `sway -d` to get a more detailed log."
I displayed a top sway version (`pacman -Qs sway`) and twice as much attached a log file. (I guess you had centuries lead that is more convenient to insert the log message)

"Tell us what wlc version you're using"
`wlc 0.0.8-2 `

"Tell us what distro you're using"
Arch Linux (beat this information in the logs files)

"XDG_RUNTIME_DIR is unset, so sway/wlc can't locate the wayland socket that lives there in order to connect to the server."
I added the `export XDG_RUNTIME_DIR=/tmp` to `~/.xinitrc`

after log:
```
04/15/17 15:16:16 - [main.c:49] [wlc] Doing work on SUID/SGID side and dropping permissions
04/15/17 15:16:16 - [main.c:49] [wlc] Work done, dropping permissions and checking communication
04/15/17 15:16:16 - [main.c:334] Dropping extra capabilities
04/15/17 15:16:16 - [main.c:338] **Failed to drop extra capabilities**
04/15/17 15:16:16 - [main.c:49] [wlc] Parent exit (20914)
```

"... and try to post a single detailed report instead of 6 comments within 5 minutes, it makes a bad impression. :)"
it is panic, Im sorry

"Hope this helps."
not really, but thanks for the answer

hook from Arch Linux (https://github.com/SirCmpwn/sway/wiki/Sway-without-systemd-on-Arch-Linux) it is complete garbage (do not misunderstand) if he needs why not include it in the release ?. 

I tried to replace my i3wm to sway even six months ago, but there were many problems with reading my config. I thought from the time the work has improved, but ...

**I propose to close this issue**, after half an hour infusions run I lost the desire to engage with the sway.
 We are waiting for better times.

**Comment by SirCmpwn** [comment]
Well, I was *going* to help this guy, but on second thought I think I won't.

**Comment by Artlant** [comment]
Perhaps the reason was the accusation of uselessness of your 'hook'. Or my English curve.

It is interesting that your defiant message appeared only after the closure of this issue. What hints 'I wanted to help you, but I changed my mind' - a useless message indicating your pettiness.
I could say with the most success, 'what to do application that will run, and will not require digging in the muck.' But I did not tell you. As it may hurt others sway developers, who do not talk in vain, and really help and correct application.

**Comment by SirCmpwn** [comment]
Alright, well, .xinitrc is not used by sway. Set the env before starting sway.

**Comment by embik** [comment]
@Artlant: Additionally, if I understand correctly, you do not need that pacman hook you complain about anyway. The wiki page says (pretty clearly) 

> If you run an Arch Linux based distribution **without** systemd [...]

But you said you're running Arch Linux, which comes _with_ systemd. You don't need it. Don't blame someone else just because you did not read the documentation properly. Of course it's useless that way.

**Comment by JerziKaminsky** [comment]
@Artlant, rudeness actually carries very well across language barriers. :man_shrugging:

**Comment by JerziKaminsky** [comment]
If anyone friendly can reproduce, please open a new issue so if it's real it can be fixed. Seems related lack of systemd?

**Comment by hobarrera** [comment]
> hook from Arch Linux (https://github.com/SirCmpwn/sway/wiki/Sway-without-systemd-on-Arch-Linux) it is complete garbage (do not misunderstand) if he needs why not include it in the release ?.

Running Arch without systemd isn't supported. If you're hacking the OS, and running it with some unsupported configuration, you can expect lots of things not to work and lots of missing configuration files out of the box.

FWIW, @JerziKaminsky, the Arch sway package, both for the stable release and the git-based one, *however* the logs posted here say he's running `ArchBang`, not `ArchLinux`, so there may be some fault there (I'm unaware of the differences).
